cmd/daemon/import.go

```diff
@@ -0,0 +1,162 @@
+package main
+
+import (
+	"fmt"
+	"github.com/gofrs/flock"
+	"github.com/pactus-project/pactus/cmd"
+	"github.com/pactus-project/pactus/genesis"
+	"github.com/pactus-project/pactus/util"
+	"github.com/spf13/cobra"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+)
+
+func buildImportCmd(parentCmd *cobra.Command) {
+	importCmd := &cobra.Command{
+		Use:   "import",
+		Short: "download and import pruned data",
+	}
+	parentCmd.AddCommand(importCmd)
+
+	workingDirOpt := addWorkingDirOption(importCmd)
+
+	importCmd.Run = func(c *cobra.Command, args []string) {
+		workingDir, err := filepath.Abs(*workingDirOpt)
+		cmd.FatalErrorCheck(err)
+
+		err = os.Chdir(workingDir)
+		cmd.FatalErrorCheck(err)
+
+		conf, gen, err := cmd.MakeConfig(workingDir)
+		cmd.FatalErrorCheck(err)
+
+		lockFilePath := filepath.Join(workingDir, ".pactus.lock")
+		fileLock := flock.New(lockFilePath)
+
+		locked, err := fileLock.TryLock()
+		if err != nil {
+			// handle unable to attempt to acquire lock
+			cmd.FatalErrorCheck(err)
+		}
+
+		if !locked {
+			cmd.PrintWarnMsgf("Could not lock '%s', another instance is running?", lockFilePath)
+
+			return
+		}
+
+		cmd.PrintInfoMsgf("Checking data in %s exists...", workingDir)
+		storeDir, _ := filepath.Abs(conf.Store.StorePath())
+		if !util.IsDirNotExistsOrEmpty(storeDir) {
+			cmd.PrintErrorMsgf("The data directory is not empty: %s", conf.Store.StorePath())
+
+			return
+		}
+
+		snapshotUrl := ""
+
+		switch gen.ChainType() {
+		case genesis.Mainnet:
+			snapshotUrl = cmd.SnapshotBaseUrl + "mainnet/"
+		case genesis.Testnet:
+			snapshotUrl = cmd.SnapshotBaseUrl + "testnet/"
+		default:
+			cmd.PrintErrorMsgf("Unsupported chain type: %s", gen.ChainType())
+
+			return
+		}
+
+		cmd.PrintInfoMsgf("Getting snapshots metadata...")
+		metadata, err := cmd.SnapshotMetadata(c.Context(), snapshotUrl)
+		if err != nil {
+			cmd.PrintErrorMsgf("Failed to get snapshot metadata: %s", err)
+
+			return
+		}
+
+		sort.Slice(metadata, func(i, j int) bool {
+			return i > j
+		})
+
+		cmd.PrintLine()
+
+		for i, m := range metadata {
+			fmt.Printf("%d. snapshot %s (%.2f MB)\n", i+1,
+				m.CreatedAt,
+				float64(m.TotalSize)/1024/1024)
```

Comment:
make a function for this in util. It will be sued in GUI later.

```go
func FormatBytesToHumanReadable(bytes uint64) string {
	const (
		_ = iota
		KB = 1 << (10 * iota)
		MB
		GB
		TB
	)
	unit := ""
	value := float64(bytes)

	switch {
	case bytes >= TB:
		unit = "TB"
		value /= TB
	case bytes >= GB:
		unit = "GB"
		value /= GB
	case bytes >= MB:
		unit = "MB"
		value /= MB
	case bytes >= KB:
		unit = "KB"
		value /= KB
	default:
		unit = "Bytes"
	}

	return fmt.Sprintf("%.2f %s", value, unit)
}

func TestFormatBytesToHumanReadable(t *testing.T) {
	tests := []struct {
		bytes    uint64
		expected string
	}{
		{1048576, "1.00 MB"},
		{3145728, "3.00 MB"},
		{1024, "1.00 KB"},
		{512, "512.00 Bytes"},
	}

	for _, test := range tests {
		result := FormatBytesToHumanReadable(test.bytes)
		if result != test.expected {
			t.Errorf("FormatBytesToHumanReadable(%d) returned %s, expected %s", test.bytes, result, test.expected)
		}
	}
}
```

Test it by yourself, I used ChatGPT for this code.